src/conf_mode/system_frr.py

```diff
@@ -51,15 +56,27 @@ def verify(frr_config):
 
 def generate(frr_config):
     # read daemons config file
-    daemons_config_current = read_file(config_file)
+    daemons_config_current = read_file(daemon_config_file)
     # generate new config file
     daemons_config_new = render_to_string('frr/daemons.frr.tmpl', frr_config)
     # update configuration file if this is necessary
     if daemons_config_new != daemons_config_current:
         syslog.warning('FRR daemons configuration file need to be changed')
-        write_file(config_file, daemons_config_new)
+        write_file(daemon_config_file, daemons_config_new)
         frr_config['config_file_changed'] = True
 
+    # profile could be automatically generated by frr in frr.conf
+    # and needs to be updated as it is taking precedence
+    if 'frrender' in frr_config and 'profile' in frr_config['frrender']:
```

Comment:
SHould be splitted into generate and apply - see https://github.com/vyos/vyos-1x/blob/current/src/conf_mode/interfaces_ethernet.py#L358-L365